### PR TITLE
Sentient floorbots can now rip up tiles when emagged + cleanup

### LINKED ISSUE #81955: Hacked Floorbot Unable to Remove Floors
Reporting client version: 515.1633

## Issue Summary

Hacked Floorbot does not/can not remove floors. Specifically when player-controlled.

## Round ID:
[225393](https://scrubby.melonmesa.com/round/225393)

## Testmerges:
- [Temp keybinds patch for 515.1628](https://www.github.com/tgstation/tgstation/pull/81089)

## Reproduction:

To reproduce, be a ghost, take control of a floorbot, have the AI disable safeties on the floorbot, you then cannot remove ANY floors despite being hacked. Also I'm quite certain that when I hack floorbots as AI, they don't remove floors then either.

### PR DESCRIPTION
**Pull request by SmArtKar**
## About The Pull Request

Closes #81955
Also performed a bit of a cleanup and added early returns everywhere to reduce the amount of pyramid shapes in the code

## Changelog
:cl:
fix: Sentient floorbots can now rip up tiles when emagged
code: Slight cleanup in floorbot code
/:cl:

### CODE REVIEW

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 1]
```diff
@@ -326,57 +328,69 @@
...
-				F.burnt = FALSE
-				F.update_appearance()
-			else if(istype(is_this_maints, /area/station/maintenance))	//place catwalk if it's plating and we're in maints
-				F.place_on_top(/turf/open/floor/catwalk_floor, flags = CHANGETURF_INHERIT_AIR)
-			else	//place normal tile if it's plating anywhere else
-				F = F.make_plating(TRUE) || F
-				F.place_on_top(/turf/open/floor/iron, flags = CHANGETURF_INHERIT_AIR)
-
-	if(!QDELETED(src))
+	var/turf/open/floor/F = target_turf
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/bot/floorbot.dm:351]: `floor` would be a better var name.

File: code/modules/mob/living/simple_animal/bot/floorbot.dm [hunk 2]
```diff
@@ -399,10 +413,19 @@
 /mob/living/simple_animal/bot/floorbot/UnarmedAttack(atom/A, proximity_flag, list/modifiers)
```
Comment (Ghommie) [inline code/modules/mob/living/simple_animal/bot/floorbot.dm:413]: Likewise, rename this var to `atom`.

**Review by Ghommie — APPROVED** [review]
Just a couple things because the code is old.

**Comment by Stiix72** [comment]
Replacing a catwalk leads to an error tile. Otherwise, floorbot is now worth using.

**Pull request merged**